### DISCUSSION TITLE
Lesson 13: Shouldn't we actually define another price_feed_address in test_set_price_feed_contract()

### DISCUSSION THREAD

**Discussion by sevamove** [smartcontractkit/full-blockchain-solidity-course-py#534] [Ideas]
Hello there,

We are currently at 13:53:27 timeline of the course with @PatrickAlphaC where we are performing [`test_set_price_feed_contract()`](https://github.com/PatrickAlphaC/defi-stake-yield-brownie-freecode/blob/2a5a1524a015abc8a1ee1fd0425a24c7b7a9a126/tests/unit/test_token_farm.py#L13) function:
```py
def test_set_price_feed_contract():
    """Test setPriceFeedContract function."""

    # Arrange.

    # Defining network.
    if network.show_active() not in LOCAL_BLOCKCHAIN_ENVIRONMENTS:
        pytest.skip("Only for local testing!")
    # Defining account.
    account = get_account()
    # Defining a different account to check onlyOwner functions.
    non_owner = get_account(index=1)
    # Deploying our main contracts.
    token_farm, dapp_token = deploy_token_farm_and_dapp_token()

    # Act.

    # Defining price feed address of ETH/USD.
    price_feed_address = get_contract("eth_usd_price_feed") #       <-----
    # Set price feed contract address to 
    token_farm.setPriceFeedContract(
        # Note: second parameter is an address and brownie knows it.
        dapp_token.address, price_feed_address, {"from": account},
    )

    # Assert.

    assert token_farm.tokenPriceFeedMapping(dapp_token.address) == price_feed_address

```

Shouldn't we actually assign the argument `"dai_usd_price_feed"` into `get_contract()` function in `price_feed_address` variable ? The idea is concluded from this line in [`deploy.py`](https://github.com/PatrickAlphaC/defi-stake-yield-brownie-freecode/blob/2a5a1524a015abc8a1ee1fd0425a24c7b7a9a126/scripts/deploy.py#L28). Of course it does not matter too much in this case whether `eth_usd_price_feed` or `dai_usd_price_feed` because they are both using the same `MockV3Aggregator` in [`helpful_scripts.py`](https://github.com/PatrickAlphaC/defi-stake-yield-brownie-freecode/blob/2a5a1524a015abc8a1ee1fd0425a24c7b7a9a126/scripts/helpful_scripts.py#L24). Glad to hear any thoughts.

**Comment by cromewar** [comment]
hello @moonwake769 well you are right, we should use `"dai_usd_price_feed"`, but I think we did `eth_usd_price_feed` as we are on ETH blockchain, so is preferably to manage all the payments and calculations using the same currency as we use for the gas payment, this makes the calculations easier.

And as you said for testing purposes it does not matter so much as both are using the MockV3Aggregator.

Best regards, Cromewar.
